packages/core/src/subscriber/chainSubscribers.ts

```diff
@@ -0,0 +1,181 @@
+import {
+    services,
+    utils,
+    config,
+    contracts,
+    database,
+} from '../../';
+import { ethers } from 'ethers';
+
+class ChainSubscribers {
+    provider: ethers.providers.WebSocketProvider;
+    ifaceCommunityAdmin: ethers.utils.Interface;
+    filterTopics: string[][];
+
+    constructor(provider: ethers.providers.WebSocketProvider) {
+        this.provider = provider;
+        this.ifaceCommunityAdmin = new ethers.utils.Interface(
+            contracts.CommunityAdminABI
+        );
+
+        this.filterTopics = [
+            [
+                ethers.utils.id(
+                    'CommunityAdded(address,address[],uint256,uint256,uint256,uint256,uint256,uint256,uint256)'
+                ),
+                ethers.utils.id('CommunityRemoved(address)'),
+            ],
+        ];
+        this.recover();
+    }
+
+    stop() {
+        this.provider.removeAllListeners();
+        services.app.ImMetadataService.setRecoverBlockUsingLastBlock();
+    }
+
+    recover() {
+        this._setupListener(this.provider);
+        // we start the listener alongside with the recover system
+        // so we know we don't lose events.
+        services.app.ImMetadataService.getRecoverBlock().then((block) =>
+            this._runRecoveryTxs(block, this.provider).then(() =>
+                services.app.ImMetadataService.removeRecoverBlock()
+            )
+        );
+    }
+
+    async _runRecoveryTxs(
+        startFromBlock: number,
+        provider: ethers.providers.JsonRpcProvider
+    ) {
+        utils.Logger.info('Recovering past events...');
+        const rawLogs = await provider.getLogs({
+            fromBlock: startFromBlock,
+            toBlock: 'latest',
+            topics: this.filterTopics,
+        });
+
+        const logs = rawLogs.sort((a, b) => {
+            if (a.blockNumber > b.blockNumber) {
+                return 1;
+            }
+            if (a.blockNumber < b.blockNumber) {
+                return -1;
+            }
+            // a must be equal to b
+            return 0;
+        });
+
+        // iterate
+        for (let x = 0; x < logs.length; x += 1) {
+            // verify if cusd or community and do things
+            await this._filterAndProcessEvent(provider, logs[x]);
+        }
+        utils.Logger.info('Past events recovered successfully!');
+    }
+
+    _setupListener(provider: ethers.providers.JsonRpcProvider) {
+        utils.Logger.info('Starting subscribers...');
+        const filter = {
+            topics: this.filterTopics,
+        };
+
+        provider.on(filter, async (log: ethers.providers.Log) => {
+            await this._filterAndProcessEvent(provider, log);
+            services.app.ImMetadataService.setLastBlock(log.blockNumber);
```

Comment:
Lets refactor this to update the last block only each 16560 (23*60*12) blocks, roughly the number of blocks each 23 hours (so it's not always at the same time in the day). We should save this number of course, but maybe save it on redis and each 23 hours save it to the database.